### PR TITLE
support 0d tensor for interpolate

### PR DESCRIPTION
**Pull request by tink2123**
### PR types

New features
### PR changes

Ops
### Describe

- support 0d tensor for intepolate att "scale" "scale_factor"
- add unittest for test_zero_dim_tensor and test_interpolate_op

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 1]
```diff
@@ -1388,6 +1388,52 @@ def test_atan2(self):
         self.assertEqual(x1.grad.numpy(), 0.5)
         self.assertEqual(x2.grad.numpy(), 0)
 
+    def test_interpolate(self):
+        import numpy as np
+
+        from paddle.nn.functional import interpolate
+
+        input_data = np.random.random((2, 3, 6, 6)).astype("float32")
+        input_x = paddle.to_tensor(input_data)
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: 可以直接 `paddle.rand([2, 3, 6, 6])`

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 2]
```diff
@@ -1388,6 +1388,52 @@ def test_atan2(self):
...
+        out3 = interpolate(
+            x=input_x,
+            scale_factor=scale_2,
+            mode="bilinear",
+            align_corners=False,
+        )
+
+        out1.backward()
+
+        np.testing.assert_allclose(
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:1447]: 加一下前向out1、out2、out3的shape测试吧
反向x.grad的shape测试

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 3]
```diff
@@ -2224,6 +2270,39 @@ def test_atan2(self):
...
+        ]
+        out1 = interpolate(
+            x=input_x, size=output_size, mode="bilinear", align_corners=False
+        )
+
+        paddle.static.append_backward(out1)
+        prog = paddle.static.default_main_program()
+        res1 = self.exe.run(prog, feed={}, fetch_list=[out1])
+
+        scale_1 = np.array(2)
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: 可以用 paddle.full([], 2)

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 4]
```diff
@@ -2224,6 +2270,39 @@ def test_atan2(self):
...
+            paddle.full([], 12, dtype="int32"),
+            paddle.full([], 12, dtype="int32"),
+        ]
+        out1 = interpolate(
+            x=input_x, size=output_size, mode="bilinear", align_corners=False
+        )
+
+        paddle.static.append_backward(out1)
+        prog = paddle.static.default_main_program()
+        res1 = self.exe.run(prog, feed={}, fetch_list=[out1])
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: fetch下x.grad_name，加下反向shape的测试吧

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py [hunk 5]
```diff
@@ -2224,6 +2270,39 @@ def test_atan2(self):
...
+            scale_factor=scale_1,
+            mode="bilinear",
+            align_corners=False,
+        )
+        paddle.static.append_backward(out2)
+        prog = paddle.static.default_main_program()
+        res2 = self.exe.run(prog, feed={}, fetch_list=[out2])
+
+        self.assertEqual(res1[0].shape, (2, 3, 12, 12))
+        self.assertEqual(res2[0].shape, (2, 3, 12, 12))
```
Comment (zhwesky2010) [inline python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:2324]: 静态图再加个case吧，0D的list：scale_0d = [paddle.full([], 2), paddle.full([], 2)]

File: python/paddle/nn/functional/common.py [hunk 6]
```diff
@@ -478,8 +484,10 @@ def _is_list_or_turple_(data):
                 else:
                     out_shape = list(out_shape)
                 for i, dim in enumerate(out_shape):
-                    if isinstance(dim, Variable):
+                    if isinstance(dim, Variable) and dim.shape != []:
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/common.py]: 可以这么写，`dim.numpy().item() ` 就覆盖[1]和[]两种情况了

File: tools/codestyle/clang_format.hook [hunk 7]
```diff
@@ -5,11 +5,11 @@ readonly VERSION="13.0.0"
 
 version=$(clang-format -version)
 
-if ! [[ $(python -V 2>&1 | awk '{print $2}' | awk -F '.' '{print $1$2}') -ge 36 ]]; then
-    echo "clang-format installation by pip need python version great equal 3.6,
-          please change the default python to higher version."
-    exit 1
-fi
+#if ! [[ $(python -V 2>&1 | awk '{print $2}' | awk -F '.' '{print $1$2}') -ge 36 ]]; then
```
Comment (zhwesky2010) [inline tools/codestyle/clang_format.hook]: 这个为什么改呀

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 1 above)
Comment (tink2123) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: done, thx

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 3 above)
Comment (tink2123) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py]: done, thx

File: tools/codestyle/clang_format.hook (hunk 7 above)
Comment (tink2123) [reply tools/codestyle/clang_format.hook]: 误修改，已恢复～

File: python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py (hunk 5 above)
Comment (tink2123) [reply python/paddle/fluid/tests/unittests/test_zero_dim_tensor.py:2324]: 静态图的scale tensor输入暂不支持多维

File: paddle/phi/infermeta/multiary.cc [hunk 8]
```diff
@@ -1550,19 +1552,25 @@ static void Interpolate2DInferShapeCheck(
...
             scale_tensor_dim.size()));
-    PADDLE_ENFORCE_EQ(scale_tensor_dim[0] == 2 || scale_tensor_dim[0] == 1,
-                      true,
-                      phi::errors::InvalidArgument(
-                          "Scale's shape must be 2 or 1, but got shape = %d .",
-                          scale_tensor_dim[0]));
+
+    if (scale_tensor_dim.size() == 1) {
+      PADDLE_ENFORCE_EQ(
+          scale_tensor_dim[0] == 2 || scale_tensor_dim[0] == 1,
```
Comment (zhwesky2010) [inline paddle/phi/infermeta/multiary.cc:1567]: 什么情况下scale的shape会是[2]呀

**Review by zhwesky2010 — APPROVED** [review]
LGTM
